nats/js/client.py

```diff
@@ -271,17 +274,36 @@ async def cb(msg):
             consumer_info = await self._jsm.add_consumer(stream, config=config)
             consumer = consumer_info.name
 
+        if consumer is None:
+            raise TypeError("cannot detect consumer")
+        if config is None:
+            raise TypeError("config is required for existing durable consumer")
+        return await self.subscribe_bind(
+            cb=cb,
+            stream=stream,
+            config=config,
+            manual_ack=manual_ack,
+            ordered_consumer=ordered_consumer,
+            consumer=consumer,
+        )
+
+    async def subscribe_bind(
```

Comment:
Think we need a test just for when using `subscribe_bind`, using `add_consumer` to create a durable consumer then bind to it to receive messages.

Reply:
Could you have a look at https://github.com/nats-io/nats.py/pull/274/commits/4d83ef72814c6c25b4fc211d002498e6c2671c42 , please? The subscriber doesn't receive any messages and dies with TimeoutError and I'm not sure how to debug it. Any advice?

Reply:
Thanks, will take a look

Reply:
Thank you. Just in case, I've added you write access to the fork. This time, I forgot to create a branch for PR, so you'll need explicit access to write into `main`.

Reply:
Hi @orsinium, I've just merged the branch with the fixed test at https://github.com/nats-io/nats.py/commit/d644c581560b838acada65a75c40cf07e8cdb369
What was needed is to setup the `deliver_subject` as part of the consumer config so that bind would succeed later.